xla/service/gpu/runtime/gpublas_lt_matmul.cc

```diff
@@ -103,12 +103,24 @@ absl::Status DoMatmul(
         c, bias));
   }));
 
-  // Get the matmul plan for this instance of matmul.
-  TF_ASSIGN_OR_RETURN(auto plan, matmul_plan.GetOrCreate([&] {
-    return ToAbsl(se::gpu::BlasLt::GetMatmulPlan(stream, *config, epilogue));
+  // Get the matmul plan for this instance of matmul: by default we allocate it 
+  // for 8 devices (which should be enough for most cases).
+  TF_ASSIGN_OR_RETURN(auto planVec, matmul_plan.GetOrCreate([] {
+    return MatmulPlanVec(8); 
```

Comment:
I got the internal comment on this:

> Instead of using 8, could we just use devID+1? Then we wouldn't need the resizing logic, right?

Reply:
Thank you for the review, the point is this code will be run for each StreamExecutor present in the system but the body of **GetOrCreate** lambda-function is executed only **once** (by whichever StreamExecutor comes first).
Suppose, we have a system with 8 GPUs, and the first one comes the executor for device_ordinal() = 3. Then, the vector MatmulVec will only contain 3 entries. Hence, anyway, resizing will be necessary later on.

I have just preallocated it for 8 GPUs (which is typically the maximal amount of GPUs present in one system). But of course, if the code is run on a larger GPU cluster, resizing will still be necessary.

Reply:
After thinking about it for a while - I found that there was a problem with my implementation: namely, ther was a racing condition when accessing **planVec**, that is why I have adapted it once again by adding mutex to MatmulPlanVec structure.